flake.nix

```diff
@@ -0,0 +1,184 @@
+
+    cabalPackages = pkgs: hpkgs:
+      inputs.nixpkgs.lib.concatMapAttrs
+      (name: value:
+        if builtins.elem name excludedPackages
+        then {}
+        else let
+          v1 =
+            if builtins.elem name noCheckPackages
+            then pkgs.haskell.lib.dontCheck value
+            else value;
+          v2 =
+            if builtins.elem name noHaddockPackages
+            then pkgs.haskell.lib.dontHaddock v1
+            else v1;
+        in {"${name}" = v2;})
+      (inputs.self.lib.cabalProject2nix ./cabal.project pkgs hpkgs (old: {
+        nativeBuildInputs = old.nativeBuildInputs ++ [pkgs.graphviz];
+      }));
+  in
+    {
+      overlays = {
+        default =
+          inputs.self.lib.overlayHaskellPackages
+          supportedGhcVersions
+          inputs.self.overlays.haskell;
+
+        haskell = inputs.self.lib.haskellOverlay cabalPackages;
+
+        ## Only needed if you are using `concat-hardware`.
+        netlist-default =
+          inputs.self.lib.overlayHaskellPackages
+          supportedGhcVersions
+          inputs.self.overlays.netlist-haskell;
+
+        netlist-haskell = import ./nix/netlist-overlay.nix;
+      };
+
+      homeConfigurations =
+        builtins.listToAttrs
+        (builtins.map
+          (system: {
+            name = "${system}-example";
+            value = inputs.home-manager.lib.homeManagerConfiguration {
+              pkgs = import inputs.nixpkgs {
+                inherit system;
+                overlays = [inputs.self.overlays.default];
+              };
+
+              modules = [
+                ({pkgs, ...}: {
+                  home.packages = [
+                    (pkgs.haskellPackages.ghcWithPackages (hpkgs: [
+                      hpkgs.concat-plugin
+                    ]))
+                  ];
+
+                  ## These attributes are simply required by home-manager.
+                  home = {
+                    homeDirectory = /tmp/${pname}-example;
+                    stateVersion = "22.11";
+                    username = "${pname}-example-user";
+                  };
+                })
+              ];
+            };
+          })
+          inputs.flake-utils.lib.defaultSystems);
+
+      ## TODO: Pull this into its own flake, for use across Haskell projects.
+      lib = import ./nix/lib.nix {inherit (inputs) bash-strict-mode nixpkgs;};
+    }
+    // inputs.flake-utils.lib.eachSystem inputs.flake-utils.lib.allSystems (system: let
+      pkgs = import inputs.nixpkgs {
+        inherit system;
+        ## NB: This uses `inputs.self.overlays.default` because packages need to
+        ##     be able to find other packages in this flake as dependencies.
+        overlays = [
+          inputs.self.overlays.netlist-default
+          inputs.self.overlays.default
+        ];
+      };
+    in {
+      packages =
+        {default = inputs.self.packages.${system}.ghc902_all;}
```

Comment:
Can we make a separate variable saying `defaultCompiler = "ghc902"` and remove this hard-coded GHC version here?